qiskit/transpiler/passes/scheduling/dynamical_decoupling.py

```diff
@@ -94,57 +94,93 @@ def uhrig_pulse_location(k):
+        dd_sequence: List[Gate],
+        qubits: Optional[List[int]] = None,
+        spacing: Optional[List[float]] = None,
+        skip_reset_qubits: bool = True,
+        pulse_alignment: int = 1,
+        extra_slack_distribution: str = "middle",
+    ):
         """Dynamical decoupling initializer.
 
         Args:
-            durations (InstructionDurations): Durations of instructions to be
-                used in scheduling.
-            dd_sequence (list[Gate]): sequence of gates to apply in idle spots.
-            qubits (list[int]): physical qubits on which to apply DD.
+            durations: Durations of instructions to be used in scheduling.
+            dd_sequence: Sequence of gates to apply in idle spots.
+            qubits: Physical qubits on which to apply DD.
                 If None, all qubits will undergo DD (when possible).
-            spacing (list[float]): a list of spacings between the DD gates.
+            spacing: A list of spacings between the DD gates.
                 The available slack will be divided according to this.
                 The list length must be one more than the length of dd_sequence,
                 and the elements must sum to 1. If None, a balanced spacing
                 will be used [d/2, d, d, ..., d, d, d/2].
-            skip_reset_qubits (bool): if True, does not insert DD on idle
-                periods that immediately follow initialized/reset qubits (as
-                qubits in the ground state are less susceptile to decoherence).
+            skip_reset_qubits: If True, does not insert DD on idle periods that
+                immediately follow initialized/reset qubits
+                (as qubits in the ground state are less susceptile to decoherence).
+            pulse_alignment: The hardware constraints for gate timing allocation.
+                This is usually provided from ``backend.configuration().timing_constraints``.
+                If provided, the delay length, i.e. ``spacing``, is implicitly adjusted to
+                satisfy this constraint.
+            extra_slack_distribution: The option to control the behavior of DD sequence generation.
+                The duration of the DD sequence should be identical to an idle time in the
+                scheduled quantum circuit, however, the delay in between gates comprising the sequence
+                should be integer number in units of dt, and it might be furter truncated
+                when ``pulse_alignment`` is specified. This sometimes results in the duration of
+                created sequence is shorter than the idle time that you want to fill with the sequence,
+                i.e. `extra slack`. This option takes following values.
+
+                    - "middle": Put the extra slack to the interval at the middle of the sequence.
+                    - "split_edges": Divide the extra slack as evenly as possible into
+                      intervals at begging and end of the sequence.
```

Comment:
```suggestion
                    - "split_edges": Divide the extra slack as evenly as possible into
                      intervals at beginning and end of the sequence.
```

Reply:
i think `edges` is probably easier (since user has to type this)

Reply:
Make sense. Done in https://github.com/Qiskit/qiskit-terra/pull/7760/commits/997049f8a4088107bd0f61ca52f1711435248273